### PR TITLE
[avmfritz] Reenable itests

### LINKED ISSUE #5413: [avmfritz] Integration tests fail
With the changes in https://github.com/openhab/openhab2-addons/pull/5412 the AVM Fritz tests now run (there used to be jaxb dependency issues). 

However most tests fail due to NPEs like:

```
TEST validDECTRepeater100Result(org.openhab.binding.avmfritz.internal.discovery.AVMFritzDiscoveryServiceTest) <<< ERROR: null
java.lang.NullPointerException
	at org.openhab.binding.avmfritz.internal.discovery.AVMFritzDiscoveryServiceTest.validDECTRepeater100Result(AVMFritzDiscoveryServiceTest.java:144)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at junit.framework.JUnit4TestAdapter.run(JUnit4TestAdapter.java:38)
	at junit.framework.TestSuite.runTest(TestSuite.java:252)
	at junit.framework.TestSuite.run(TestSuite.java:247)
	at aQute.junit.Activator.test(Activator.java:340)
	at aQute.junit.Activator.automatic(Activator.java:236)
	at aQute.junit.Activator.run(Activator.java:177)
	at aQute.launcher.Launcher.lambda$serviceChanged$0(Launcher.java:1385)
	at aQute.launcher.Launcher.run(Launcher.java:352)
	at aQute.launcher.Launcher.main(Launcher.java:152)
```

and 

```
TEST validSwitchGroupDiscoveryResult(org.openhab.binding.avmfritz.internal.discovery.AVMFritzDiscoveryServiceTest) <<< ERROR: null
java.lang.NullPointerException
	at org.openhab.binding.avmfritz.internal.discovery.AVMFritzDiscoveryServiceTest.validSwitchGroupDiscoveryResult(AVMFritzDiscoveryServiceTest.java:797)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at junit.framework.JUnit4TestAdapter.run(JUnit4TestAdapter.java:38)
	at junit.framework.TestSuite.runTest(TestSuite.java:252)
	at junit.framework.TestSuite.run(TestSuite.java:247)
	at aQute.junit.Activator.test(Activator.java:340)
	at aQute.junit.Activator.automatic(Activator.java:236)
	at aQute.junit.Activator.run(Activator.java:177)
	at aQute.launcher.Launcher.lambda$serviceChanged$0(Launcher.java:1385)
	at aQute.launcher.Launcher.run(Launcher.java:352)
	at aQute.launcher.Launcher.main(Launcher.java:152)

```

### PR DESCRIPTION
**Pull request by cweitkamp**
- Changed dependencies based on Java EE modules (openhab/openhab-core#920)
- Reactivated avmfritz tests
- Fixed Nest tests

Adresses #5775
Fixes #5413

I disabled non working itests and told Travis to run with performing itests. Another issue popped up related to Astro issues (seems to be related to #5720).

```
[ERROR] Tests run: 22, Failures: 9, Errors: 0, Skipped: 1, Time elapsed: 0.338 s <<< FAILURE! - in org.openhab.binding.astro.internal.calc.SunCalcTest
[ERROR] testGetSunInfoForNauticDawnAccuracy(org.openhab.binding.astro.internal.calc.SunCalcTest)  Time elapsed: 0.059 s  <<< FAILURE!
java.lang.AssertionError: expected:<1.55124826E12> but was:<1.55124485E12>
	at org.openhab.binding.astro.internal.calc.SunCalcTest.testGetSunInfoForNauticDawnAccuracy(SunCalcTest.java:109)
[ERROR] testGetSunInfoForNauticDuskAccuracy(org.openhab.binding.astro.internal.calc.SunCalcTest)  Time elapsed: 0.011 s  <<< FAILURE!
java.lang.AssertionError: expected:<1.55129335E12> but was:<1.55128981E12>
	at org.openhab.binding.astro.internal.calc.SunCalcTest.testGetSunInfoForNauticDuskAccuracy(SunCalcTest.java:169)
[ERROR] testGetSunInfoForAstronomicalDawnAccuracy(org.openhab.binding.astro.internal.calc.SunCalcTest)  Time elapsed: 0.01 s  <<< FAILURE!
java.lang.AssertionError: expected:<1.5512459E12> but was:<1.55124236E12>
	at org.openhab.binding.astro.internal.calc.SunCalcTest.testGetSunInfoForAstronomicalDawnAccuracy(SunCalcTest.java:98)
[ERROR] testGetSunInfoForAstronomicalDuskAccuracy(org.openhab.binding.astro.internal.calc.SunCalcTest)  Time elapsed: 0.011 s  <<< FAILURE!
java.lang.AssertionError: expected:<1.55129571E12> but was:<1.55129217E12>
	at org.openhab.binding.astro.internal.calc.SunCalcTest.testGetSunInfoForAstronomicalDuskAccuracy(SunCalcTest.java:180)
[ERROR] testGetSunInfoForRiseAccuracy(org.openhab.binding.astro.internal.calc.SunCalcTest)  Time elapsed: 0.006 s  <<< FAILURE!
java.lang.AssertionError: expected:<1.55125272E12> but was:<1.55124918E12>
	at org.openhab.binding.astro.internal.calc.SunCalcTest.testGetSunInfoForRiseAccuracy(SunCalcTest.java:131)
[ERROR] testGetSunInfoForCivilDawnAccuracy(org.openhab.binding.astro.internal.calc.SunCalcTest)  Time elapsed: 0.006 s  <<< FAILURE!
java.lang.AssertionError: expected:<1.55125062E12> but was:<1.55124708E12>
	at org.openhab.binding.astro.internal.calc.SunCalcTest.testGetSunInfoForCivilDawnAccuracy(SunCalcTest.java:120)
[ERROR] testGetSunInfoForCivilDuskAccuracy(org.openhab.binding.astro.internal.calc.SunCalcTest)  Time elapsed: 0.011 s  <<< FAILURE!
java.lang.AssertionError: expected:<1.55129125E12> but was:<1.55128771E12>
	at org.openhab.binding.astro.internal.calc.SunCalcTest.testGetSunInfoForCivilDuskAccuracy(SunCalcTest.java:158)
[ERROR] testGetSunInfoForSetAccuracy(org.openhab.binding.astro.internal.calc.SunCalcTest)  Time elapsed: 0.01 s  <<< FAILURE!
java.lang.AssertionError: expected:<1.55129125E12> but was:<1.55128758E12>
	at org.openhab.binding.astro.internal.calc.SunCalcTest.testGetSunInfoForSetAccuracy(SunCalcTest.java:149)
[ERROR] testGetSunInfoForSunNoonAccuracy(org.openhab.binding.astro.internal.calc.SunCalcTest)  Time elapsed: 0.004 s  <<< FAILURE!
java.lang.AssertionError: expected:<1.55127199E12> but was:<1.55126845E12>
	at org.openhab.binding.astro.internal.calc.SunCalcTest.testGetSunInfoForSunNoonAccuracy(SunCalcTest.java:140)
[INFO] Running org.openhab.binding.astro.internal.model.SunTest
[INFO] Tests run: 17, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.001 s - in org.openhab.binding.astro.internal.model.SunTest
[INFO] 
[INFO] Results:
[INFO] 
[ERROR] Failures: 
[ERROR]   SunCalcTest.testGetSunInfoForAstronomicalDawnAccuracy:98 expected:<1.5512459E12> but was:<1.55124236E12>
[ERROR]   SunCalcTest.testGetSunInfoForAstronomicalDuskAccuracy:180 expected:<1.55129571E12> but was:<1.55129217E12>
[ERROR]   SunCalcTest.testGetSunInfoForCivilDawnAccuracy:120 expected:<1.55125062E12> but was:<1.55124708E12>
[ERROR]   SunCalcTest.testGetSunInfoForCivilDuskAccuracy:158 expected:<1.55129125E12> but was:<1.55128771E12>
[ERROR]   SunCalcTest.testGetSunInfoForNauticDawnAccuracy:109 expected:<1.55124826E12> but was:<1.55124485E12>
[ERROR]   SunCalcTest.testGetSunInfoForNauticDuskAccuracy:169 expected:<1.55129335E12> but was:<1.55128981E12>
[ERROR]   SunCalcTest.testGetSunInfoForRiseAccuracy:131 expected:<1.55125272E12> but was:<1.55124918E12>
[ERROR]   SunCalcTest.testGetSunInfoForSetAccuracy:149 expected:<1.55129125E12> but was:<1.55128758E12>
[ERROR]   SunCalcTest.testGetSunInfoForSunNoonAccuracy:140 expected:<1.55127199E12> but was:<1.55126845E12>
[INFO] 
[ERROR] Tests run: 39, Failures: 9, Errors: 0, Skipped: 1
```

I am not sure if the Hue tests are really working because they are complaining about unsatisfied references too.
```
Components implementing org.eclipse.smarthome.config.discovery.DiscoveryService:
5 [ACTIVE] org.openhab.binding.hue.internal.discovery.HueBridgeNupnpDiscovery in org.openhab.binding.hue
46 [UNSATISFIED_REFERENCE] org.eclipse.smarthome.config.discovery.upnp.internal.UpnpDiscoveryService in org.openhab.core.config.discovery.upnp
	UpnpService (org.jupnp.UpnpService)
		1 [UNSATISFIED_REFERENCE] org.jupnp.UpnpServiceImpl in org.jupnp
			OSGiUpnpServiceConfiguration (org.jupnp.OSGiUpnpServiceConfiguration)
			HttpService (org.osgi.service.http.HttpService)
				No component implementing org.osgi.service.http.HttpService is currently registered.
Make sure to add the appropriate bundle and set 'Default Auto-Start=true' in the launch config.
Components implementing org.eclipse.smarthome.config.discovery.DiscoveryService:
5 [ACTIVE] org.openhab.binding.hue.internal.discovery.HueBridgeNupnpDiscovery in org.openhab.binding.hue
46 [UNSATISFIED_REFERENCE] org.eclipse.smarthome.config.discovery.upnp.internal.UpnpDiscoveryService in org.openhab.core.config.discovery.upnp
	UpnpService (org.jupnp.UpnpService)
		1 [UNSATISFIED_REFERENCE] org.jupnp.UpnpServiceImpl in org.jupnp
			OSGiUpnpServiceConfiguration (org.jupnp.OSGiUpnpServiceConfiguration)
			HttpService (org.osgi.service.http.HttpService)
				No component implementing org.osgi.service.http.HttpService is currently registered.
Make sure to add the appropriate bundle and set 'Default Auto-Start=true' in the launch config.
```

Signed-off-by: Christoph Weitkamp <github@christophweitkamp.de>

### CODE REVIEW

**Comment by J-N-K** [comment]
The astro one is very ... interesting. I have seen that in a local build, too. Next two builds succeeded, then it failed again. No idea why it works sometimes and sometimes not.

We could probably fix it by increasing the tolerance of the comparison. But I didn‘t check if it is really a wrong calculation when it fails.

**Comment by cweitkamp** [comment]
We had this Java 11 Travis build error again. I have the slight feeling that it could be related to Travis caching of the `.m2` folder.

**Comment by cweitkamp** [comment]
I reverted testing stuff and opened #5834 for the astro test issues. Maybe this at least helps to finalize #5751.

File: bundles/org.openhab.binding.avmfritz/src/main/feature/feature.xml [hunk 1]
```diff
@@ -5,10 +5,9 @@
     <feature name="openhab-binding-avmfritz" description="AVM FRITZ!Box Binding" version="${project.version}">
         <feature>openhab-runtime-base</feature>
         <feature>openhab-transport-upnp</feature>
-        <bundle dependency="true">mvn:org.apache.servicemix.specs/org.apache.servicemix.specs.activation-api-1.1/2.9.0</bundle>
-        <bundle dependency="true">mvn:javax.xml.bind/jaxb-api/2.2.12</bundle>
-        <bundle dependency="true">mvn:com.sun.xml.bind/jaxb-impl/2.2.11</bundle>
-        <bundle dependency="true">mvn:com.sun.xml.bind/jaxb-core/2.2.11</bundle>
+        <bundle>mvn:org.apache.servicemix.specs/org.apache.servicemix.specs.activation-api-1.1/2.9.0</bundle>
```
Comment (J-N-K) [inline bundles/org.openhab.binding.avmfritz/src/main/feature/feature.xml]: Did you remove the `dependency="true"` intentionally?

Comment (cweitkamp) [reply bundles/org.openhab.binding.avmfritz/src/main/feature/feature.xml]: No. Added it again.

File: bundles/org.openhab.binding.avmfritz/src/main/feature/feature.xml [hunk 2]
```diff
@@ -6,9 +6,8 @@
         <feature>openhab-runtime-base</feature>
         <feature>openhab-transport-upnp</feature>
         <bundle dependency="true">mvn:org.apache.servicemix.specs/org.apache.servicemix.specs.activation-api-1.1/2.9.0</bundle>
-        <bundle dependency="true">mvn:javax.xml.bind/jaxb-api/2.2.12</bundle>
-        <bundle dependency="true">mvn:com.sun.xml.bind/jaxb-impl/2.2.11</bundle>
-        <bundle dependency="true">mvn:com.sun.xml.bind/jaxb-core/2.2.11</bundle>
+        <bundle dependency="true">mvn:org.apache.servicemix.specs/org.apache.servicemix.specs.jaxb-api-2.2/2.9.0</bundle>
```
Comment (maggu2810) [inline bundles/org.openhab.binding.avmfritz/src/main/feature/feature.xml]: There had been features created for such stuff, so no need to depend on bundles (use the feature):
https://github.com/openhab/openhab-core/pull/920/files#diff-ed272408a699f2280d7706c517dc3c53R111-R122

Comment (cweitkamp) [reply bundles/org.openhab.binding.avmfritz/src/main/feature/feature.xml]: :facepalm: Yes, of course. I will change it here and for some other bindings too.

Comment (cweitkamp) [reply bundles/org.openhab.binding.avmfritz/src/main/feature/feature.xml]: Done. See #5878.

**Pull request merged**